### ISSUE TITLE
DTMF

### ISSUE THREAD

**Issue by atulgo** [cristiansteib/Sim800l#31]
Detection of DTMF tones  in my SIM800L module.

I get "OK" when I send 'AT+DDET=1,1000,1,1* ' over uart . 
But it doesnt seem to detect anything afterwards.
Perhaps I am doing something wrong. 
Would be great to get some feedback.

regards
Atul
ref : *SIM800 Series_AT Command Manual_V1.09

**Comment by ankitghevariya** [comment]
@atulgo 

Hiii,

I tried to do the same what you written.

I send command to GSM like this ` AT+DDET=1,100,1,1\r\n` over UART

I was getting ERROR when I have not inserted SIM in GSM. But after inserting SIM I send the same command and got `OK` response.

Then I dialed call and when opposite party picked the call and press the numbers from keypad of mobile phone, What response I got shown below.

``` C++
+DTMF: 1,40

+DTMF: 2,100

+DTMF: 3,120

+DTMF: 6,140

+DTMF: 7,340

+DTMF: 8,360

+DTMF: 9,380

+DTMF: *,200

+DTMF: 0,140

+DTMF: #,300

+DTMF: 1,340
```
The first data shows the digit/character pressed from the mobile phone on call with GSM, and second data shows how long time that character was pressed.

I hope this information will solve your problem

**Comment by atulgo** [comment]
Will try to replicate ur steps today.
Thanks a lot.:)

**Comment by Mamdouh-Freelancer** [comment]
I have a question my phone doesn't receive DTMF tones while the call (can't be heard) should SIM800L or DTMF decoder module work with it?

**Comment by ankitghevariya** [comment]
@Mamdouh-Freelancer 

Yes, you will get DTMF data in SIM800L. Could you please tell me in detail, what testing you have made so I can guide you properly.

**Comment by Mamdouh-Freelancer** [comment]
> @Mamdouh-Freelancer
> 
> Yes, you will get DTMF data in SIM800L. Could you please tell me in detail, what testing you have made so I can guide you properly.

Thanks for your response, Below is what I tried with normal phone call using old/smart phones:

1. I tried to call from mobile to a landline and DTMF worked well in landline but also not with all mobiles.
2. Tried many troubleshooting like in/decreasing volume and check after reading this article also I checked this and this.
3. Tried to disable dial pad tones in android and  calling any IVR service  and It still work with no Issues.
4. Tried to call cross net or on net mobile numbers and there is no new.
5. Tried to enable DTMF detection on GSM phone using AT+VTD? but it returns ERROR as same as SIM900's AT command AT+DDET.

After long googling times I found that phone companies may put all the signaling on a different channel not through voice channel. I wish if DTMF decoder can decodes too (as same as SIM800L) incoming signals through that different channel? Is this forbidden only in my country's mobile operators so?

I tried this project [DTMF Decoder Using Only Arduino © GPL3+  DTMF decoder without an 8870)](http://%22https//create.arduino.cc/projecthub/MM_Shoaib/dtmf-decoder-using-only-arduino-872502?f=1%22). But it doesn't work properly as serial monitor is printing out even no sounds and never prints the right digits pressed. I hope anyone can clarify more about the diagram (for electronic beginners like me) and connection, specially capacitor position If it needs a new post I will post ASAP.

**Comment by Rogerjoumblat** [comment]
Esse módulo não serve para nada as operadoras de telefonia bloqueiam Sim800l

**Comment by Mark-Wills** [comment]
Not in the UK - GPRS is still going. However it is set for sunsetting at
some point in the future.

**Comment by MohamedESalem** [comment]
@Mamdouh-Freelancer Hello mamdouh i hope you are doing well.
I am facing the same problem with the sim800l's DTMF tones, Kindly i wish to know what did you come up with after about two years
